lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -238,30 +219,47 @@ class TraceElements extends Gatherer {
   }
 
   /**
-   * @param {LH.Gatherer.PassContext} passContext
+   * @param {LH.Gatherer.FRTransitionalContext} context
    */
-  async beforePass(passContext) {
-    await passContext.driver.sendCommand('Animation.enable');
+  async startInstrumentation(context) {
+    await context.driver.defaultSession.sendCommand('Animation.enable');
+
+    /** @param {LH.Crdp.Animation.AnimationStartedEvent} args */
+    this.onAnimationCreated = ({animation: {id, name}}) => {
+      name && this.animationIdToName.set(id, name);
+    };
+
+    context.driver.defaultSession.on('Animation.animationStarted', this.onAnimationCreated);
   }
 
   /**
-   * @param {LH.Gatherer.PassContext} passContext
-   * @param {LH.Gatherer.LoadData} loadData
+   * @param {LH.Gatherer.FRTransitionalContext} context
+   */
+  async stopInstrumentation(context) {
+    if (this.onAnimationCreated) {
```

Comment:
could avoid this `if` by moving `onAnimationCreated` to class property next to the map, but understand if you'd rather keep the impl inside `startInstrumentation` :)

Reply:
why not a private function?

Reply:
Needed access to `this`, I wound up defining it in a new constructor.

Reply:
See JSUsage gatherer constructor for how to use `this` in class callback functions.